common:
    dataset: "ouster"
    lid_topic:  "/ouster/points"
    imu_topic:  "/ouster/imu"
    time_sync_en: false         # ONLY turn on when external time synchronization is really not possible

preprocess:
    lidar_type: 3                # 1 for Livox serials LiDAR, 2 for Velodyne LiDAR, 3 for ouster LiDAR,
    scan_line: 64
    blind: 1
    time_scale: 1e-3

mapping:
    acc_cov: 0.2
    gyr_cov: 0.2
    b_acc_cov: 0.002
    b_gyr_cov: 0.002
    fov_degree:    180
    det_range:     150.0
    extrinsic_est_en:  true      # true: enable the online estimation of IMU-LiDAR extrinsic
    extrinsic_T: [ 0.0, 0.0, 0.0 ]
    extrinsic_R: [1, 0, 0,
                  0, 1, 0,
                  0, 0, 1]

rtk_trans:
    extrinsic_T: [ 0.0, 0.0, 0.0]
    extrinsic_R: [ 0.0, 0.0, 22.957]

publish:
    scan_publish_en:  true       # false: close all the point cloud output
    dense_publish_en: false       # false: low down the points number in a global-frame point clouds scan.
    scan_bodyframe_pub_en: true  # true: output the point cloud scans in IMU-body-frame

path_save_en: true                 # 保存轨迹，用于精度计算和比较
output_debug_en: false

pcd_save:
    pcd_save_en: false
    interval: -1                 # how many LiDAR frames saved in each pcd file;
    # -1 : all frames will be saved in ONE pcd file, may lead to memory crash when having too much frames.

feature_extract_enable: false
point_filter_num: 6
max_iteration: 3
filter_size_surf: 0.5
filter_size_map: 0.5
cube_side_length: 1000

nearest_neighbor_type: 1 # 1 for IKD_TREE, 2 for IVOX
ivox_grid_resolution: 0.5        # default=0.2
ivox_nearby_type: 18             # 6, 18, 26
esti_plane_threshold: 0.1        # default=0.1

loop_closure:
    loop_closure_frequency: 100
    graph_iter_num: 50   # graph optimization iterations
    history_time_threshold: 60.0  # seconds, history_time_threshold
    lp_distance_threshold: 50.0   # meters, the distance between too keyframe
    lp_ndt_trans_score_threshold: 5.0   # the score of between two scan-map by ndt align
    lp_ndt_fitness_score_threshold: 1.0  # the fitness score of between two scan-map by ndt align
    lp_pose_noise: 0.01   #
    lp_angle_noise: 0.05   #
    kf_num_sub_map: 20  # the keyframe number that add to sub_map
    min_id_threshold: 20  # 被选为候选的两个关键帧之间的ID差值
    skip_id: 10   #如果选择了一个候选帧，那么隔开多少个ID之后再选一个
    loop_sparse_enable: true  # default true, false
